core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -2765,6 +2765,34 @@ public void testDeleteRetentionMsBeingCancelledBeforeSecondDelete() throws Remot
         verify(remoteStorageManager).deleteLogSegmentData(metadataList.get(1));
     }
 
+    @Test
+    public void testDeleteRetentionMsBiggerThanTimeMs() throws RemoteStorageException, ExecutionException, InterruptedException {
+        LogConfig mockLogConfig = new LogConfig(Map.of("retention.ms", time.milliseconds() + 100));
```

Comment:
If, let's say, test is running on a slow machine and by the time we reach to build `RetentionTimeData`, 100ms has already elapsed since the start of the test, in those cases, our cleanupUntilMs will actually become positive, leading to a flaky test.

Could we set this value to a large value such as 1 month after current epoch?

Reply:
Yes, set it to 1 month after current epoch. Thanks for the suggestion 👍 .